frontend/src/redux/actions/actions.ts

```diff
@@ -41,16 +42,18 @@ export const getUserRejected = (error: Error): GetUserAction => ({
 
 export const detectUser = (): ThunkAction<void, AppState, unknown, Action<string>> => {
```

Comment:
I get detectUser is a thunk -- but perhaps we need to turn it into an effect, that uses a dispatch to trigger it after it comes back. That way we can clean up the effect when it gets remounted / recalled